customer/api/sales/views.py

```diff
@@ -6,13 +7,56 @@
 
 
 class CustomerViewSet(AuditLoggingModelViewSet):
+    SEARCH_VALUE_MIN_LENGTH = 2
+
     queryset = Customer.objects.all().order_by(
         "primary_profile__last_name", "primary_profile__first_name"
     )
     serializer_class = CustomerSerializer
     permission_classes = [permissions.AllowAny]
     http_method_names = ["get", "post", "put", "head"]  # disable PATCH
 
+    def get_queryset(self):
+        if self.get_serializer_class() is CustomerListSerializer:
+            first_name = self.request.query_params.get("first_name", "")
+            last_name = self.request.query_params.get("last_name", "")
+            phone_number = self.request.query_params.get("phone_number", "")
+            email = self.request.query_params.get("email", "")
+            search_values_less_than_min_length = (
+                len(first_name) < self.SEARCH_VALUE_MIN_LENGTH
+                and len(last_name) < self.SEARCH_VALUE_MIN_LENGTH
+                and len(phone_number) < self.SEARCH_VALUE_MIN_LENGTH
+                and len(email) < self.SEARCH_VALUE_MIN_LENGTH
+            )
+            if search_values_less_than_min_length:
+                return Customer.objects.none()
+
+            queryset = Customer.objects.all().order_by(
+                "primary_profile__last_name", "primary_profile__first_name"
+            )
+            if first_name:
+                queryset = queryset.filter(
+                    Q(primary_profile__first_name__contains=first_name)
```

Comment:
I think we want these filters to be case-insensitive, so these should be `icontains`.

Reply:
That is true. Fixed.